crates/kubelet/src/state.rs

```diff
@@ -254,44 +257,49 @@ pub async fn run_to_completion<PodState: Send + Sync + 'static>(
     client: &kube::Client,
     state: impl State<PodState>,
     pod_state: &mut PodState,
-    pod: &Pod,
+    pod: Arc<RwLock<Pod>>,
 ) {
-    let api: Api<KubePod> = Api::namespaced(client.clone(), pod.namespace());
+    let initial_pod = { pod.read().await.clone() };
+    let namespace = initial_pod.namespace().to_string();
+    let name = initial_pod.name().to_string();
+    let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
+
+    // Initialize Pod Container Statuses
+    patch_status(&api, &name, make_registered_status(&initial_pod)).await;
```

Comment:
I have implemented my suggestion of applying this initial status update prior to applying the Provider's first status. I also removed the use of this function in the Providers to verify that this eliminates their responsibility to do this.